lib/galaxy_social.py

```diff
@@ -0,0 +1,196 @@
+import json
+import os
+import sys
+from argparse import ArgumentParser
+from fnmatch import filter
+from importlib import import_module
+
+from bs4 import BeautifulSoup
+from jsonschema import validate
+from markdown import markdown
+from yaml import safe_load as yaml
+
+
+class galaxy_social:
+    def __init__(self, preview: bool, json_out: str):
+        self.preview = preview
+        self.json_out = json_out
+        plugins_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), "..", "plugins.yml"
+        )
+        with open(plugins_path, "r") as file:
+            self.plugins_config = yaml(file)
+
+        self.plugins = {}
+        for plugin in self.plugins_config["plugins"]:
+            if preview and plugin["name"].lower() != "markdown":
+                continue
+
+            if plugin["enabled"]:
+                module_name, class_name = plugin["class"].rsplit(".", 1)
+                try:
+                    module_path = f"{'lib.' if not os.path.dirname(os.path.abspath(sys.argv[0])).endswith('lib') else ''}plugins.{module_name}"
+                    module = import_module(module_path)
+                    plugin_class = getattr(module, class_name)
+                except Exception as e:
+                    raise Exception(
+                        f"Error with plugin {module_name}.{class_name}.\n{e}"
+                    )
+
+                try:
+                    config = {}
+                    if plugin.get("config"):
+                        for key, value in plugin["config"].items():
+                            if isinstance(value, str) and value.startswith("$"):
+                                if os.environ.get(value[1:]):
+                                    config[key] = os.environ.get(value[1:])
+                                else:
+                                    raise Exception(
+                                        f"Missing environment variable {value[1:]}."
+                                    )
+                            else:
+                                config[key] = value
+                except Exception as e:
+                    raise Exception(
+                        f"Missing config for {module_name}.{class_name}.\n{e}"
+                    )
```

Comment:
This will also catch the inner exception and be misleading, but this looks outdated anyway since you're using .get("config") up above. Likely no try/except here, but an else.